<center>
<img src="images/airbnb.png" style="width:150px;">
</center>
<h1 style="font-family: Trebuchet MS; padding: 12px; font-size: 48px; color: #FF5A5F; text-align: center; line-height: 1.25;"><b>Airbnb Booking Analysis</b></h1>
<hr style="border: 1px solid #FF5A5F; background-color: #FF5A5F; width: 100%;">


<h1 style="font-family: Trebuchet MS; padding: 12px; font-size: 32px; color: #FF5A5F; line-height: 1.25;"><b>Purpose of the Analysis :</b></h1>


<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;">This analysis aims to uncover the factors influencing Airbnb prices in New York City, providing insights into the relationships and patterns within various variables. By doing so, our study offers valuable information for both travelers and hosts, along with key insights for Airbnb's business strategy.<span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>Data Exploration and Cleaning Process :</b><span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;">The project began with a thorough exploration and cleaning of the dataset to ensure its readiness for analysis. We started by examining the dataset’s characteristics—such as data types, missing values, and distribution of values—to understand its structure. The cleaning process involved handling inconsistencies, missing values, and any duplicate records, as well as removing outliers. This preparation stage was essential to ensure data quality, which helps prevent potential biases and errors that could impact our findings.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>Preparing the Data for Analysis :</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;">Following the initial exploration and cleaning, we finalized the dataset to ensure high quality, making it ready for analysis. This step is crucial, as it provides a solid foundation for accurate results and reliable insights. With the data now ready, we could dive into answering specific research questions, free from distractions caused by errors or inconsistencies.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>Data Exploration and Pattern Identification :</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;">With the dataset cleaned and structured, we proceeded to explore and summarize it through descriptive statistics and visualizations. These exploratory steps helped us uncover patterns and trends, as well as relationships among various factors. By developing these insights, we aimed to better understand how different variables interact, the underlying causes of trends, and other influencing factors.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>Visualizing Patterns in the Data :</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;">Using data visualization, we explored Airbnb data patterns and trends by creating a range of graphs and charts. Observations and insights were documented alongside each visualization to further explain the findings. This visual approach allowed us to identify meaningful patterns, such as factors influencing prices and availability across neighborhoods.</span>

<h1 style="font-family: Trebuchet MS; padding: 12px; font-size: 32px; color: #FF5A5F; line-height: 1.25;"><b>Problem Statements :</b></h1>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>1. What are the most popular neighborhoods for Airbnb rentals in New York City? How do prices and availability differ across neighborhoods ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>2. How has New York City's Airbnb market evolved over time? Are there notable trends in the number of listings, pricing, or occupancy rates ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>3. Are there any clear patterns in the types of properties rented on Airbnb in New York City? Are certain property types more popular or tend to be priced higher ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>4. Which factors appear to correlate with Airbnb rental prices in New York City ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>5. Which areas in New York City offer the best opportunities for property investment at reasonable prices with high booking demand ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>6. How do the lengths of stay for Airbnb rentals vary by neighborhood in New York City? Are there neighborhoods that attract longer stays more than others ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>7. Is there a relationship between Airbnb rental ratings and prices in New York City? Are higher-priced listings more likely to have better ratings ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>8. What are the total and maximum number of reviews for each neighborhood group ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>9. Which room types are the most reviewed within each neighborhood group on a monthly basis ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>10. Which listings or locations offer the best experience for travelers in New York City ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>11. Which locations or properties are ideal for hosts in New York City ?</b></span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>12. How do prices vary across different neighborhood groups in New York City ?</b></span>

<h1 style="font-family: Trebuchet MS; padding: 12px; font-size: 32px; color: #FF5A5F; line-height: 1.25;"><b>Understanding Given Variables :</b></h1>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>id:</b> A unique identifier assigned to each listing in the dataset.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>name:</b> The title of the listing, as it appears on Airbnb.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>host_id:</b> A unique identifier assigned to each host in the dataset.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>host_name:</b> The name of the host, as displayed on Airbnb.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>neighbourhood_group:</b> The broader neighborhood area within New York City, such as Manhattan or Brooklyn.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>neighbourhood:</b> The specific neighborhood location of the listing.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>latitude:</b> The geographical latitude coordinate of the listing.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>longitude:</b> The geographical longitude coordinate of the listing.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>room_type:</b> The type of accommodation offered, such as an entire home, private room, or shared room.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>price:</b> The nightly rental price for the listing, in US dollars.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>minimum_nights:</b> The minimum number of nights a guest is required to book.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>number_of_reviews:</b> The total number of reviews the listing has received.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>reviews_per_month:</b> The average number of reviews the listing receives each month.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>calculated_host_listings_count:</b> The total number of listings that a host has on Airbnb.</span>

<span style="font-family: Trebuchet MS; font-size: 18px; color: #FF5A5F; line-height: 1.25;"><b>availability_365:</b> The number of days the listing is available for booking in the upcoming year (365 days).</span>

<h1 style="font-family: Trebuchet MS; padding: 12px; font-size: 32px; color: #FF5A5F; line-height: 1.25;"><b>Import Libraries</b></h1>

In [42]:
import warnings
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as plt

%matplotlib inline
warnings.filterwarnings('ignore')

<h1 style="font-family: Trebuchet MS; padding: 12px; font-size: 32px; color: #FF5A5F; line-height: 1.25;"><b>Data Exploration & Data Cleaning</b></h1>

In [43]:
airbnb_df = pd.read_csv("dataset/airbnb_nyc_2019.csv")

In [44]:
airbnb_df.head().T

0  \
id                                                            2539   
name                            Clean & quiet apt home by the park   
host_id                                                       2787   
host_name                                                     John   
neighbourhood_group                                       Brooklyn   
neighbourhood                                           Kensington   
latitude                                                  40.64749   
longitude                                                -73.97237   
room_type                                             Private room   
price                                                          149   
minimum_nights                                                   1   
number_of_reviews                                                9   
last_review                                             2018-10-19   
reviews_per_month                                             0.21   
calculated_host_listings_count                                   6   
availability_365                                               365   

                                                    1  \
id                                               2595   
name                            Skylit Midtown Castle   
host_id                                          2845   
host_name                                    Jennifer   
neighbourhood_group                         Manhattan   
neighbourhood                                 Midtown   
latitude                                     40.75362   
longitude                                   -73.98377   
room_type                             Entire home/apt   
price                                             225   
minimum_nights                                      1   
number_of_reviews                                  45   
last_review                                2019-05-21   
reviews_per_month                                0.38   
calculated_host_listings_count                      2   
availability_365                                  355   

                                                                  2  \
id                                                             3647   
name                            THE VILLAGE OF HARLEM....NEW YORK !   
host_id                                                        4632   
host_name                                                 Elisabeth   
neighbourhood_group                                       Manhattan   
neighbourhood                                                Harlem   
latitude                                                   40.80902   
longitude                                                  -73.9419   
room_type                                              Private room   
price                                                           150   
minimum_nights                                                    3   
number_of_reviews                                                 0   
last_review                                                     NaN   
reviews_per_month                                               NaN   
calculated_host_listings_count                                    1   
availability_365                                                365   

                                                              3  \
id                                                         3831   
name                            Cozy Entire Floor of Brownstone   
host_id                                                    4869   
host_name                                           LisaRoxanne   
neighbourhood_group                                    Brooklyn   
neighbourhood                                      Clinton Hill   
latitude                                               40.68514   
longitude                                             -73.95976   
room_type                                       Entire home/apt   
price                     

<span style="font-family: Trebuchet MS; font-size: 24px; color: #FF5A5F; line-height: 1.25;">Now, rename few columns to better understanding of variables</span>

In [45]:
rename_col = {'id':'listing_id','name':'listing_name','number_of_reviews':'total_reviews','calculated_host_listings_count':'host_listings_count'}

In [48]:
airbnb_df = airbnb_df.rename(columns = rename_col)
airbnb_df.head().T

0  \
listing_id                                         2539   
listing_name         Clean & quiet apt home by the park   
host_id                                            2787   
host_name                                          John   
neighbourhood_group                            Brooklyn   
neighbourhood                                Kensington   
latitude                                       40.64749   
longitude                                     -73.97237   
room_type                                  Private room   
price                                               149   
minimum_nights                                        1   
total_reviews                                         9   
last_review                                  2018-10-19   
reviews_per_month                                  0.21   
host_listings_count                                   6   
availability_365                                    365   

                                         1  \
listing_id                            2595   
listing_name         Skylit Midtown Castle   
host_id                               2845   
host_name                         Jennifer   
neighbourhood_group              Manhattan   
neighbourhood                      Midtown   
latitude                          40.75362   
longitude                        -73.98377   
room_type                  Entire home/apt   
price                                  225   
minimum_nights                           1   
total_reviews                           45   
last_review                     2019-05-21   
reviews_per_month                     0.38   
host_listings_count                      2   
availability_365                       355   

                                                       2  \
listing_id                                          3647   
listing_name         THE VILLAGE OF HARLEM....NEW YORK !   
host_id                                             4632   
host_name                                      Elisabeth   
neighbourhood_group                            Manhattan   
neighbourhood                                     Harlem   
latitude                                        40.80902   
longitude                                       -73.9419   
room_type                                   Private room   
price                                                150   
minimum_nights                                         3   
total_reviews                                          0   
last_review                                          NaN   
reviews_per_month                                    NaN   
host_listings_count                                    1   
availability_365                                     365   

                                                   3  \
listing_id                                      3831   
listing_name         Cozy Entire Floor of Brownstone   
host_id                                         4869   
host_name                                LisaRoxanne   
neighbourhood_group                         Brooklyn   
neighbourhood                           Clinton Hill   
latitude                                    40.68514   
longitude                                  -73.95976   
room_type                            Entire home/apt   
price                                             89   
minimum_nights                                     1   
total_reviews                                    270   
last_review                               2019-07-05   
reviews_per_month                               4.64   
host_listings_count                                1   
availability_365                                 194   

                                                                    4  
listing_id                                                       5022  
listing_name         Entire Apt: Spacious Studio/Loft by central park  
host_id                                                          7192  
host_

<span style="font-family: Trebuchet MS; font-size: 24px; color: #FF5A5F; line-height: 1.25;">Checking shape of Dataset</span>

In [49]:
airbnb_df.shape

(48895, 16)

<span style="font-family: Trebuchet MS; font-size: 24px; color: #FF5A5F; line-height: 1.25;">Basic information of Dataset</span>

In [50]:
airbnb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   listing_id           48895 non-null  int64  
 1   listing_name         48879 non-null  object 
 2   host_id              48895 non-null  int64  
 3   host_name            48874 non-null  object 
 4   neighbourhood_group  48895 non-null  object 
 5   neighbourhood        48895 non-null  object 
 6   latitude             48895 non-null  float64
 7   longitude            48895 non-null  float64
 8   room_type            48895 non-null  object 
 9   price                48895 non-null  int64  
 10  minimum_nights       48895 non-null  int64  
 11  total_reviews        48895 non-null  int64  
 12  last_review          38843 non-null  object 
 13  reviews_per_month    38843 non-null  float64
 14  host_listings_count  48895 non-null  int64  
 15  availability_365     48895 non-null 

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>NOTE :</b></span>

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;">So, host_name, neighbourhood_group, neighbourhood and room_type fall into categorical variable category.</span>
    
<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;">While host_id, latitude, longitude, price, minimum_nights, number_of_reviews, last_review, reviews_per_month, host_listings_count, availability_365 are numerical variables</span>

<span style="font-family: Trebuchet MS; font-size: 24px; color: #FF5A5F; line-height: 1.25;">Check duplicate row in Dataset</span>

In [51]:
airbnb_df = airbnb_df.drop_duplicates()
airbnb_df.count()

listing_id             48895
listing_name           48879
host_id                48895
host_name              48874
neighbourhood_group    48895
neighbourhood          48895
latitude               48895
longitude              48895
room_type              48895
price                  48895
minimum_nights         48895
total_reviews          48895
last_review            38843
reviews_per_month      38843
host_listings_count    48895
availability_365       48895
dtype: int64

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>NOTE :</b></span>
<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;">So, there is no duplicate rows</span>

<span style="font-family: Trebuchet MS; font-size: 24px; color: #FF5A5F; line-height: 1.25;">Checking NULL values in Dataset</span>

In [52]:
airbnb_df.isnull().sum()

listing_id                 0
listing_name              16
host_id                    0
host_name                 21
neighbourhood_group        0
neighbourhood              0
latitude                   0
longitude                  0
room_type                  0
price                      0
minimum_nights             0
total_reviews              0
last_review            10052
reviews_per_month      10052
host_listings_count        0
availability_365           0
dtype: int64

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>NOTE :</b></span>
<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>host_name</b> and <b>listing_name</b> have not large amount of NULL values, so firstly fill those with some substitutes in both the columns.</span>

In [53]:
airbnb_df['listing_name'].fillna('unknown',inplace=True)
airbnb_df['host_name'].fillna('no_name',inplace=True)

In [54]:
airbnb_df[['host_name','listing_name']].isnull().sum()

host_name       0
listing_name    0
dtype: int64

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>NOTE :</b></span>
<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;">Now, the columns <b>last_review</b> and <b>reviews_per_month</b> have total 10052 null values each.</span>
<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>last_review</b> column is not required for our analysis as compared to <b>number_of_reviews</b> & <b>reviews_per_month</b>. We're good to drop this column.</span>

In [55]:
airbnb_df = airbnb_df.drop(['last_review'], axis=1)

In [56]:
airbnb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 15 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   listing_id           48895 non-null  int64  
 1   listing_name         48895 non-null  object 
 2   host_id              48895 non-null  int64  
 3   host_name            48895 non-null  object 
 4   neighbourhood_group  48895 non-null  object 
 5   neighbourhood        48895 non-null  object 
 6   latitude             48895 non-null  float64
 7   longitude            48895 non-null  float64
 8   room_type            48895 non-null  object 
 9   price                48895 non-null  int64  
 10  minimum_nights       48895 non-null  int64  
 11  total_reviews        48895 non-null  int64  
 12  reviews_per_month    38843 non-null  float64
 13  host_listings_count  48895 non-null  int64  
 14  availability_365     48895 non-null  int64  
dtypes: float64(3), int64(7), object(5)
m

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>NOTE :</b></span>
<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;">The <b>reviews_per_month</b> column also containing NULL/NaN values and we can simple put 0 reviews by replacing NAN's</span>

In [57]:
airbnb_df['reviews_per_month'] = airbnb_df['reviews_per_month'].replace(to_replace=np.nan,value=0).astype('int64')

In [58]:
airbnb_df['reviews_per_month'].isnull().sum() 

0

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;"><b>NOTE :</b></span>
<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;">So, there is no null value now in <b>reviews_per_month</b> column because we replaced NULL/NaN value by 0.</span>

<span style="font-family: Trebuchet MS; font-size: 20px; color: #FF5A5F; line-height: 1.25;">Now check the dataset with no NULL values and totally cleaned and ready to analysis.</span>

In [60]:
airbnb_df.sample(5).T

17565  \
listing_id                                           13814049   
listing_name         Subway accessible apartment in Brooklyn!   
host_id                                              81329032   
host_name                                              Ashlee   
neighbourhood_group                                  Brooklyn   
neighbourhood                                      Kensington   
latitude                                             40.64231   
longitude                                           -73.97974   
room_type                                        Private room   
price                                                      50   
minimum_nights                                              5   
total_reviews                                               0   
reviews_per_month                                           0   
host_listings_count                                         1   
availability_365                                            0   

                                                            614    \
listing_id                                                 232612   
listing_name         Lovely pied-à-terre, in an historic building   
host_id                                                   1217241   
host_name                                                     Dee   
neighbourhood_group                                      Brooklyn   
neighbourhood                                  Bedford-Stuyvesant   
latitude                                                 40.68309   
longitude                                               -73.94219   
room_type                                         Entire home/apt   
price                                                         110   
minimum_nights                                                  3   
total_reviews                                                 305   
reviews_per_month                                               3   
host_listings_count                                             1   
availability_365                                              231   

                                                  25918  \
listing_id                                     20705026   
listing_name         New Clean Private Room In Flushing   
host_id                                       137129564   
host_name                                          Bing   
neighbourhood_group                              Queens   
neighbourhood                                  Flushing   
latitude                                       40.75224   
longitude                                     -73.81505   
room_type                                  Private room   
price                                                49   
minimum_nights                                        1   
total_reviews                                        77   
reviews_per_month                                     3   
host_listings_count                                   4   
availability_365                                    362   

                                                    16131  \
listing_id                                       13031854   
listing_name         Samaria's Brownstone (Crown Heights)   
host_id                                          72019599   
host_name                                           Maria   
neighbourhood_group                              Brooklyn   
neighbourhood                               Crown Heights   
latitude                                         40.67218   
longitude                                       -73.95368   
room_type                                 Entire home/apt   
price                                                 125   
minimum_nights                                          2   
total_reviews                                          70   
reviews_per_month                                       1   
host_listings_count                                     1   
availability_365                                  